AmplifyTools/amplify-tools.sh

```diff
@@ -5,15 +5,24 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-set -e
+# Check NodeJS installation
 
 if ! which node >/dev/null; then
   echo "warning: Node is not installed. Visit https://nodejs.org/en/download/ to install it"
   exit 1
 fi
 
+# Check for NVM and make sure it's initialized
+NVM_PATH=~/.nvm/nvm.sh
+if [ -f "${NVM_PATH}" ]; then
+  echo "NVM found, initializing it..."
+  source $NVM_PATH
```

Comment:
1.  If we want to support directories with spaces, this needs to be:
```
NVM_PATH="${HOME}/.nvm/nvm.sh"
```
(note that ~ can not be used in a string)

2.  Then if we assume there is a space in the path, we should change the `source` command on line 19 to have quotes like:
```
source "${NVM_PATH}"
```

3.  Btw, what does `nvm.sh` expect to have?  i'm guessing it updates some path variables?

Reply:
I don't fully understand what is inside nvm.sh,  but going to assume it is the right thing to do..

Reply:
me neither, it's a fairly long file (~3500 lines) that handles both NVM and NodeJS version initialization. Based on my research, that safe to call and there are NVM+Xcode issues out there that suggest the same thing to fix it.

I was able to repro the issue, Xcode - for some reason - doesn't pick my default NodeJS version, which is controlled by NVM. That results in the wrong version of amplify CLI being used by our script, so even when I do `npm -g install @aws-amplify/cli` in my terminal Xcode still picks the wrong version. By forcing the NVM initialization inside the Xcode Build Phase the problem is gone.